### PR TITLE
Updates to Spool File Labels in Jobs View

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

If spool file has PROCSTEP, show label as `stepname:ddname - procstep`
If no PROCSTEP, show label as `stepname:ddname - record count`

Hover over spool file label to see all attributes of the spool file

## Release Notes

Milestone: 2.3.0 or following enhancement release

Changelog: Update to Job's View Spool file label to display PROCSTEP if available, if PROCSTEP isn't available the label will display the Spool file's record count.  
Added Spool file attribute information to a hover over the Spool file's name.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

https://user-images.githubusercontent.com/49491949/186192288-e669748d-bf04-49a5-a134-879fda77715d.mov

### CODE REVIEW

File: packages/zowe-explorer/src/job/ZoweJobNode.ts [hunk 1]
```diff
@@ -211,7 +231,16 @@ export class Job extends ZoweTreeNode implements IZoweJobTreeNode {
         return this.getParent() ? this.getParent().getSessionNode() : this;
     }
 
+    set tooltip(newTooltip: string) {
+        if (newTooltip !== undefined) {
+            this._tooltip = newTooltip;
+        }
+    }
```
Comment (kristinochka) [inline packages/zowe-explorer/src/job/ZoweJobNode.ts:238]: if (newTooltip) {
     this._tooltip = newTooltip;
  }
I think this check is saver against any [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value.

File: packages/zowe-explorer/src/job/ZoweJobNode.ts [hunk 2]
```diff
@@ -211,7 +231,16 @@ export class Job extends ZoweTreeNode implements IZoweJobTreeNode {
...
+
     get tooltip(): string {
+        if (this._tooltip !== undefined) {
+            return this._tooltip;
+        }
```
Comment (kristinochka) [inline packages/zowe-explorer/src/job/ZoweJobNode.ts:243]: Same as above

**Comment by kristinochka** [comment]
I think this looks good. Could you link a ticket associated with this change?

File: packages/zowe-explorer/src/job/ZoweJobNode.ts (hunk 2 above)
Comment (JillieBeanSim) [reply packages/zowe-explorer/src/job/ZoweJobNode.ts:243]: addressed in commit https://github.com/zowe/vscode-extension-for-zowe/pull/1920/commits/1889233360272455accc66bbba7c913a16d6a1aa

File: packages/zowe-explorer/src/job/ZoweJobNode.ts (hunk 1 above)
Comment (JillieBeanSim) [reply packages/zowe-explorer/src/job/ZoweJobNode.ts:238]: addressed in commit https://github.com/zowe/vscode-extension-for-zowe/pull/1920/commits/1889233360272455accc66bbba7c913a16d6a1aa

**Review by KutluOzel-b — APPROVED** [review]
@JillieBeanSim  LGTM 👍  , thanks for this change.

**Review by kristinochka — APPROVED** [review]

**Pull request merged**